File: src/main/java/edu/telran/homework/hw03/Main.java
package edu.telran.homework.hw03;

import java.util.Arrays;
import java.util.Random;
import java.util.stream.IntStream;

public class Main {
    public static void main(String[] args) {

        System.out.println(Arrays.toString(bubbleSort(rndArr(10))));
        System.out.println(Arrays.toString(bubbleSort(rndArr(18))));
        System.out.println(Arrays.toString(bubbleSort(rndArr(100))));
    }

    private static int[] bubbleSort(int[] intArr) {
        System.out.println(Arrays.toString(intArr));
        boolean isSwapped;
        do {
            isSwapped = false;
            for (int i = 0; i < intArr.length - 1; i++) {
                if (intArr[i] > intArr[i + 1]) {
                    int tmp = intArr[i];
                    intArr[i] = intArr[i + 1];
                    intArr[i + 1] = tmp;
                    isSwapped = true;
                }
            }
        } while (isSwapped);

        return intArr;
    }

    private static int[] rndArr(int arrLength) {
        return IntStream.generate(() -> new Random().nextInt(100)).limit(arrLength).toArray();
    }
}

File: src/main/java/edu/telran/homework/hw03/main.js
const bubbleSort = intArr => {
    console.log(intArr)
    let isSwapped
    do {
        isSwapped = false
        for (let i = 0; i < intArr.length - 1; i++) {
            if (intArr[i] > intArr[i + 1]) {
                [intArr[i + 1], intArr[i]] = [intArr[i], intArr[i + 1]]
                isSwapped = true
            }
        }
    } while (isSwapped)

    return intArr
}

const testArr = (length, limit = 100) => [... new Array(length)].map(_ => Math.floor(Math.random() * limit))

console.log(bubbleSort(testArr(4)))
console.log(bubbleSort(testArr(12)))
console.log(bubbleSort(testArr(100, 10000)))
